packages/backend/migrations/20241119125038_add_unique_constraint_to_kid.js

```diff
@@ -0,0 +1,37 @@
+/**
+ * @param { import("knex").Knex } knex
+ * @returns { Promise<void> }
+ */
+exports.up = async function (knex) {
+  return knex
+    .raw(
+      // Keep only one active walletAddressKey (the most recent one)
+      `DELETE FROM "walletAddressKeys" w
+        WHERE revoked = false
+        AND "createdAt" <> (
+          SELECT MAX("createdAt")
+          FROM "walletAddressKeys"
+          WHERE revoked = false
+          AND "walletAddressId" = w."walletAddressId"
+          GROUP BY "walletAddressId"
+);
```

Comment:
This is a tricky one. One thing we need to add is the `kid` parameter (so we don't delete all non-revoked keys for a wallet address), and also maybe we can come up with a way to not delete by `createdAt` (although very unlikely, it could end up deleting valid, non-duplicated keys which happen to have the same createdAt).

Reply:
Yeah, I think that grouping by `kid` is the right way to do it.
```suggestion
      `DELETE FROM "walletAddressKeys" w
        WHERE revoked = false
        AND "createdAt" <> (
          SELECT MAX("createdAt")
          FROM "walletAddressKeys"
          WHERE revoked = false
          AND kid = w.kid
          GROUP BY kid
        );
```
Could we use `TOP` to delete just one key if there are multiple valid keys with the same `createdAt`? I don't see another way to determine which key to choose.

Reply:
This is a way to do it, but still based on `createdAt`. If there are multiple keys that have the most recent `createdAt`, we can keep the first one we find and delete the rest. What do you think, @mkurapov ?

```suggestion
      `DELETE FROM "walletAddressKeys" w
       WHERE revoked = false
       AND w."createdAt" = (
          SELECT "createdAt"
          FROM "walletAddressKeys"
          WHERE revoked = false
            AND kid = w.kid
          ORDER BY "createdAt" DESC
          LIMIT 1
       )
       AND w.id <> (
          SELECT id
          FROM "walletAddressKeys"
          WHERE revoked = false
            AND kid = w.kid
          ORDER BY "createdAt" DESC
          LIMIT 1 OFFSET 1
       );
      `
```

Reply:
@oana-lolea I was looking through the PR history, and I think we can use physical location row `ctid` value:

https://github.com/interledger/rafiki/blob/5e8156950a4806565dbb9b54d67da3cdb9fdf4b3/packages/backend/migrations/20240731084359_add_unique_constraint_to_kid.js#L8-L13

I agree with you on keeping the most recent key so we'd have to adjust the migration it a bit, but that should work.